group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupCoordinatorService.java

```diff
@@ -1580,83 +1574,51 @@ public CompletableFuture<DeleteShareGroupOffsetsResponseData> deleteShareGroupOf
-            if (topicId != null) {
-                requestTopicIdToNameMapping.put(topicId, topic.topicName());
-                deleteShareGroupStateRequestTopicsData.add(new DeleteShareGroupStateRequestData.DeleteStateData()
-                    .setTopicId(topicId)
-                    .setPartitions(
-                        topic.partitions().stream().map(
-                            partitionIndex -> new DeleteShareGroupStateRequestData.PartitionData().setPartition(partitionIndex)
-                        ).toList()
-                    ));
-            } else {
-                deleteShareGroupOffsetsResponseTopicList.add(new DeleteShareGroupOffsetsResponseData.DeleteShareGroupOffsetsResponseTopic()
-                    .setTopicName(topic.topicName())
-                    .setPartitions(topic.partitions().stream().map(
-                        partition -> new DeleteShareGroupOffsetsResponseData.DeleteShareGroupOffsetsResponsePartition()
-                            .setPartitionIndex(partition)
-                            .setErrorCode(Errors.UNKNOWN_TOPIC_OR_PARTITION.code())
-                            .setErrorMessage(Errors.UNKNOWN_TOPIC_OR_PARTITION.message())
-                    ).toList()));
-            }
-        });
-
-        // If the request for the persister is empty, just complete the operation right away.
-        if (deleteShareGroupStateRequestTopicsData.isEmpty()) {
+        if (requestData.topics() == null || requestData.topics().isEmpty()) {
             return CompletableFuture.completedFuture(
                 new DeleteShareGroupOffsetsResponseData()
-                    .setResponses(deleteShareGroupOffsetsResponseTopicList));
+            );
         }
 
-        CompletableFuture<DeleteShareGroupOffsetsResponseData> future = new CompletableFuture<>();
+        return runtime.scheduleReadOperation(
+            "share-group-delete-offsets-request",
+            topicPartitionFor(groupId),
+            (coordinator, lastCommittedOffset) -> coordinator.shareGroupDeleteOffsetsRequest(groupId, requestData)
+        )
+            .thenCompose(resultHolder -> {
+                if (resultHolder == null) {
+                    log.error("Failed to retrieve deleteState request parameters from group coordinator for the group {}", groupId);
+                    return CompletableFuture.completedFuture(
+                        DeleteShareGroupOffsetsRequest.getErrorDeleteResponseData(Errors.UNKNOWN_SERVER_ERROR)
+                    );
+                }
+
+                if (resultHolder.topLevelError().code() != Errors.NONE.code()) {
+                    log.error("Failed to retrieve deleteState request parameters from group coordinator for the group {}", groupId);
+                    return CompletableFuture.completedFuture(
+                        DeleteShareGroupOffsetsRequest.getErrorDeleteResponseData(resultHolder.topLevelError())
```

Comment:
And here's a case where you want to maintain the error message so it's available to the admin client.